### DISCUSSION TITLE
`noUnsafeArgument`, from `typescript-eslint/no-unsafe-argument`

### DISCUSSION THREAD

**Discussion by jpike88** [biomejs/biome#341] [Ideas]
It's easy to pass an any type as an argument, which opens up problems that are the result of not properly ensuring validation of the type beforehand.

https://typescript-eslint.io/rules/no-unsafe-argument/

code that would be allowed without this rule:

```
class Test {
  constructor(wtf: string) {
    console.info(wtf);
  }
}

function wtf(uhh: Promise<number>) {
  // wat
}

const a: any = 3;
const x = new Test(a)
wtf(a);
```

**Comment by Conaclos** [comment]
I wonder if [noExplicitAny](https://biomejs.dev/linter/rules/no-explicit-any) doesn't cover your cases?

**Reply by jpike88** [reply]
I'd like to enable it one day, but it doesn't really cover those cases.

Reason is that 3rd party libraries may deliberately return any (e.g. calling knex, where it could be any[] from calling a database query), and noExplicitAny doesn't detect prevent that. That's the mean reason I want those two rules so I can catch anys from trickling in from 3rd party typings.

**Reply by Conaclos** [reply]
I see. This could require type information to implement these rules.

**Comment by Conaclos** [comment]
Thanks to bring it to light! I added this rule in the [list of rules we want to support](https://github.com/biomejs/biome/discussions/3).
